conda-store-server/conda_store_server/server/views/api.py

```diff
@@ -688,6 +688,60 @@ async def api_put_build(
     }
 
 
+@router_api.put(
+    "/build/{build_id}/cancel/",
+    response_model=schema.APIAckResponse,
+)
+async def api_put_build_cancel(
+    build_id: int,
+    request: Request,
+    conda_store=Depends(dependencies.get_conda_store),
+    db: Session = Depends(dependencies.get_db),
+    auth=Depends(dependencies.get_auth),
+):
+    build = api.get_build(db, build_id)
+    if build is None:
+        raise HTTPException(status_code=404, detail="build id does not exist")
+
+    auth.authorize_request(
+        request,
+        f"{build.environment.namespace.name}/{build.environment.name}",
+        {Permissions.BUILD_CANCEL},
+        require=True,
+    )
+
+    if conda_store.celery_app.control.inspect().ping() is None:
+        raise HTTPException(
+            status_code=409,
+            detail="conda-store celery broker does not support task cancelation. Use redis or rabbitmq message queues. See docs for a more detailed explanation",
+        )
```

Comment:
The said docs are here (took me a bit to find): https://docs.celeryq.dev/en/latest/userguide/workers.html#worker-persistent-revokes
> Note that remote control commands must be working for revokes to work. Remote control commands are only supported by the RabbitMQ (amqp) and Redis at this point.